sympy/physics/continuum_mechanics/column.py

```diff
@@ -0,0 +1,330 @@
+
+        if self._end_conditions['top'] == "pinned" and self._end_conditions['bottom'] == "pinned":
+            self._boundary_conditions['deflection'].append((self._height, 0))
+
+        elif self._end_conditions['top'] == "fixed" and self._end_conditions['bottom'] == "fixed":
+            # `M` is the reaction moment
+            M = Symbol('M')
+            self._boundary_conditions['deflection'].append((self._height, 0))
+            # moment  = P*y - M
+            self._moment -= M
+
+        elif self._end_conditions['top'] == "pinned" and self._end_conditions['bottom'] == "fixed":
+            # `F` is the horizontal force at the pinned end to counter the rection moment at fixed end
+            F = Symbol('F')
+            self._boundary_conditions['deflection'].append((self._height, 0))
+            # moment = P*y - F(l - x)
+            self._moment -= F*(self.height - x)
+
+        elif self._end_conditions['top'] == "free" and self._end_conditions['bottom'] == "fixed":
+            # `d` is the deflection at the free end
+            d = Symbol('d')
+            self._boundary_conditions['deflection'].append((self._height, d))
+            # moment = P*y - P*d
+            self._moment -= P*d
+
+        else:
+            raise ValueError("{} {} end-condition is not supported".format(sstr(self._end_conditions['top']), sstr(self._end_conditions['bottom'])))
+
+
+    def solve_slope_deflection(self):
+        """
+        Solves the differnetial equation of buckling to determine the
+        deflection and slope equations.
+
+        Examples
+        ========
+
+        A column of timber section is 8m long both ends being fixed.
+        Young's modulus of the timber is E and second-moment I.
+        The applied load on the column is 360KN.
+
+        >>> from sympy.physics.continuum_mechanics.column import Column
+        >>> from sympy import Symbol, symbols
+        >>> E, I, P = symbols('E, I, P', positive=True)
+        >>> c = Column(8, E, I, 360000, top="fixed", bottom="fixed")
+        >>> c.end_conditions
+        {'bottom': 'fixed', 'top': 'fixed'}
+        >>> c.boundary_conditions
+        {'deflection': [(0, 0), (8, 0)], 'slope': [(0, 0)]}
+        >>> c.moment()
+        -M + 360000*y(x)
+        >>> c.solve_slope_deflection()
+        >>> c.deflection()
+        -M*cos(600*x/(sqrt(E)*sqrt(I)))/360000 + M/360000
+        >>> c.slope()
+        M*sin(600*x/(sqrt(E)*sqrt(I)))/(600*sqrt(E)*sqrt(I))
+        """
+        y = Function('y')
+        x = self._variable
+        P = Symbol('P', positive=True)
+
+        C1, C2 = symbols('C1, C2')
+        E = self._elastic_modulus
+        I = self._second_moment
+
+        # differnetial equation of Column buckling
+        eq = E*I*y(x).diff(x, 2) + self._moment
+
+        defl_sol = dsolve(Eq(eq, 0), y(x)).args[1]
+        slope_sol = defl_sol.diff(x)
+
+        constants = list(linsolve([defl_sol.subs(x, 0), slope_sol.subs(x, 0)], C1, C2).args[0])
```

Comment:
You should use the ics argument to dsolve to find the integration constants rather than explicitly solving for them here.

Reply:
Sure, will do it in the next commit

Reply:
I tried it with `ics` but it fails for the case `self._deflection==0` where we explicitly solve the equations with unknowns using `solve()` for (P, C1, C2).
If we use the `ics` argument at first we might need to use `dsolve()` again to obtain the equation with C1 and C2. But that might increase the run-time, as it would run dsolve twice.
Is it okay to not use the `ics`?

Reply:
I don't understand what you are saying here. If dsolve with ics doesn't work then maybe that's a bug in dsolve that should be fixed. Duplicating the code that already exists for this purpose seems like a mistake to me.

If it is a bug in dsolve then that doesn't mean you have to fix it in this PR. It just means you should open an issue for dsolve and add an XFAIL test that refers back to the dsolve issue.

Reply:
Actually, I have handled a special case (for pinned-pinned condition) where the deflection comes out to be zero after we substitute values of C1 and C2. You can see this in Line no. 233.
In this special case I have made the deflection equation to explicitly solve for C1  and C2 using solve(), in line no. 244. 
Now whether or not we use `ics` initially in line no. 223, we might require an equation with the constants(C1 and C2) for this special case, and for that we would have to use dsolve() without ics.
So if we use `ics` initially in line no. 223 we might need to again call dsolve for this special case. Therefore, I am saying it will be better to call dsolve once only, 

There is no issue with dsolve.

Reply:
I'm not sure exactly what you mean but it sounds to me that maybe when dsolve with ics returns a zero solution you can find the solution some other way algebraically and don't need to use dsolve.

Or maybe it's that the ODEs+ics don't actually have a unique solution and dsolve misses the important solution - that would be a bug in dsolve.

Either way if you can solve this by calling dsolve twice then I think that's probably better than reimplementing the IC handling code.